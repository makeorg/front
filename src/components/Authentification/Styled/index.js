import styled from 'styled-components';
import { pxToRem } from '../../../helpers/styled';
import Breakpoints from '../../../assets/vars/Breakpoints';
import { SpaceBetweenColumnToRow } from '../../Elements/FlexElements';

const Authentification = styled(SpaceBetweenColumnToRow)`
  width: 100%;
  margin-top: ${pxToRem('10px')};
  @media (min-width: ${pxToRem(Breakpoints.mobile)}){
    margin-top: ${pxToRem('15px')};
  }
`;


export default Authentification;
