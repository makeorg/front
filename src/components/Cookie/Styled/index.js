import styled from 'styled-components';
import { pxToRem } from 'Helpers/styled';
import { BackgroundColors } from 'Assets/vars/Colors';
import { UnstyledButton } from 'Components/Elements/ButtonElements';
import { Breakpoints, Layouts, DefaultPadding } from 'Assets/vars/Breakpoints';

export const Wrapper = styled.section`
  position: relative;
  display: flex;
  justify-content: center;
  width: 100%;
  background: ${BackgroundColors.LightGrey};
  background-color: ${BackgroundColors.LightGrey};
`;

export const Content = styled.p`
  width: 100%;
  max-width: ${pxToRem(Layouts.ContainerWidth)};
  padding: ${pxToRem(DefaultPadding.Mobile)} ${pxToRem('30px')} ${pxToRem(DefaultPadding.Mobile)} ${pxToRem(DefaultPadding.Mobile)};
  font-size: ${pxToRem('12px')};
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}){
    font-size: ${pxToRem('14px')};
  }  
  @media (min-width: ${pxToRem(Layouts.SpecialContainerWidth)}){
    padding: ${pxToRem(DefaultPadding.Mobile)};
  }
`;

export const CloseButton = styled(UnstyledButton)`
  position: absolute;
  top: ${pxToRem(DefaultPadding.Mobile)};
  right: ${pxToRem(DefaultPadding.Mobile)};
  z-index: 1;
  font-size: ${pxToRem('20px')};
`;
