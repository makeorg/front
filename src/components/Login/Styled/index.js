import styled from 'styled-components';
import { rem } from 'polished';
import {
  SepWrapper,
  SepText,
  Form,
  ExtraParagraph,
  ExtraAltParagraph,
  RedLinkButton
} from './Content';

const Login = styled.section`
  display: flex;
  flex-flow: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  min-height: ${rem('490px')};
  max-width: ${rem('470px')};
`;

/* Content */
Login.SepWrapper = SepWrapper;
Login.SepText = SepText;
Login.Form = Form;
Login.ExtraParagraph = ExtraParagraph;
Login.ExtraAltParagraph = ExtraAltParagraph;
Login.RedLinkButton = RedLinkButton;

export default Login;
