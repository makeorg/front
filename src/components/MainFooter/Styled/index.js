import styled from 'styled-components';
import { rem } from 'polished';
import { ShadowColors } from '../../../assets/vars/Colors';
import {
  Nav,
  Title,
  HighlightedTitle,
  Link
} from './FooterContent';

const MainFooter = styled.footer`
  padding: ${rem('15px')} ${rem('20px')};
  box-shadow: 0 0 16px 6px ${ShadowColors.BlackZeroTwoOpacity};
  background: ${props => props.theme.MainColor};
  background-color: ${props => props.theme.MainColor};
`;

/* FooterContent */
MainFooter.Nav = Nav;
MainFooter.Title = Title;
MainFooter.HighlightedTitle = HighlightedTitle;
MainFooter.Link = Link;

export default MainFooter;
