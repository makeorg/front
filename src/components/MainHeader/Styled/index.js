import styled from 'styled-components';
import { rem } from 'polished';
import { BasicColors, ShadowColors } from '../../../assets/vars/Colors';
import Logo from './HeaderContent';

const MainHeader = styled.header`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  padding: ${rem('15px')} ${rem('20px')};
  box-shadow: 0 2px 4px 0 ${ShadowColors.BlackHalfOpacity};
  background: ${BasicColors.PureWhite};
  background-color: ${BasicColors.PureWhite};
`;

/* HeaderContent */
MainHeader.Logo = Logo;

export default MainHeader;
