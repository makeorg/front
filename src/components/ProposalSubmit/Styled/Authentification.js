import styled from 'styled-components';
import { rem } from 'polished';

const ProposalSubmitAuthentificationWrapper = styled.div`
  max-width: ${rem('620px')};
  display: flex;
  flex-flow: column;
  justify-content: space-between;
  align-items: center;
  text-align: center;
`;

export default ProposalSubmitAuthentificationWrapper;
