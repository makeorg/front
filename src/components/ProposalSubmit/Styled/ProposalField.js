import styled from 'styled-components';
import { rem } from 'polished';
import Breakpoints from '../../../assets/vars/Breakpoints';
import { NoStyleTextInput } from '../../Elements/Form';
import { MiddleRow } from '../../Elements/FlexElements';
import { SmallGreyButton, SmallRedButton } from '../../Elements/ButtonElements';
import { MakeFonts } from '../../../assets/vars/Fonts';
import { TextColors } from '../../../assets/vars/Colors';

export const Label = styled.label`
  font-size: ${rem('14px')};
  line-height: ${rem('40px')};
  margin-left:  ${rem('12px')};
  min-width: ${rem('42px')};
  @media (min-width: ${rem(Breakpoints.mobile)}){
    font-size: ${rem('18px')};
    line-height: ${rem('50px')};
    margin-left: ${rem('22px')};
    min-width: ${rem('52px')};
  }
`;

export const Input = styled(NoStyleTextInput)`
  width: 100%;
  font-family: ${MakeFonts.CircularBold};
  font-size: ${rem('14px')};
  line-height: ${rem('40px')};
  padding-right: 0;
  @media (min-width: ${rem(Breakpoints.mobile)}){
    font-size: ${rem('18px')};
    line-height: ${rem('50px')};
  }
`;

export const CharLimit = styled(MiddleRow)`
  font-family: ${MakeFonts.CircularBook};
  color: ${TextColors.MediumGrey};
  font-size: ${rem('12px')};
  min-width: ${rem('55px')};
  @media (min-width: ${rem(Breakpoints.mobile)}){
    font-size: ${rem('14px')};
    min-width: ${rem('70px')};
  }
`;

export const DisabledProposalButton = styled(SmallGreyButton)`
  padding: ${rem('9px')} ${rem('20px')} ${rem('7px')};
  box-shadow: none;
`;

export const ProposalButton = styled(SmallRedButton)`
  padding: ${rem('9px')} ${rem('20px')} ${rem('7px')};
  box-shadow: none;
`;
