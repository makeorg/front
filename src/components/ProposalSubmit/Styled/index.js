import styled from 'styled-components';
import { pxToRem } from 'Helpers/styled';
import { BasicColors, BorderColors } from 'Assets/vars/Colors';
import { Breakpoints, Layouts } from 'Assets/vars/Breakpoints';
import {
  InputWrapper,
  Label,
  Input,
  ButtonWrapper,
  CharLimit,
  ProposalButton,
  DisabledProposalButton,
  ProposalIcon
} from './ProposalField';

export const DescriptionWrapper = styled.div``;

const ProposalSubmitForm = styled.form`
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  width: 100%;
  padding: 0 ${pxToRem('8px')};
  border: ${pxToRem('1px')} solid ${BorderColors.LightGrey};
  border-radius: ${pxToRem('30px')};
  background: ${BasicColors.PureWhite};
  background-color: ${BasicColors.PureWhite};
`;

export const ProposalSubmitFormWrapper = styled.aside`
  display: block;
  width: 100%;
  max-height: ${pxToRem('45px')};
  max-width: ${pxToRem(Layouts.ContainerWidth)};
  margin: ${pxToRem('15px')} auto ${pxToRem('5px')};
  padding: 0 ${pxToRem('20px')};
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}){
    max-height: ${pxToRem('50px')};
    margin: ${pxToRem('15px')} auto;
  }
`;

/* Proposal Field */
ProposalSubmitForm.InputWrapper = InputWrapper;
ProposalSubmitForm.Label = Label;
ProposalSubmitForm.Input = Input;
ProposalSubmitForm.ButtonWrapper = ButtonWrapper;
ProposalSubmitForm.CharLimit = CharLimit;
ProposalSubmitForm.ProposalButton = ProposalButton;
ProposalSubmitForm.DisabledProposalButton = DisabledProposalButton;
ProposalSubmitForm.ProposalIcon = ProposalIcon;

export default ProposalSubmitForm;
