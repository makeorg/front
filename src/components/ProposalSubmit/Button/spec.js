import { shallow } from 'enzyme';
import ProposalSubmitButtonComponent from './';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { ProposalButton, DisabledProposalButton } from '../Styled/ProposalField';

describe('ProposalSubmitButtonComponent', () => {
  const enabledButtonProps = {
    canSubmit: true
  };
  const disabledButtonProps = {
    canSubmit: false
  };

  it('Renders enabled button', () => {
    const wrapper = shallow(<ProposalSubmitButtonComponent {...enabledButtonProps} />);

    expect(wrapper.find(ProposalButton)).toHaveLength(1);
    expect(wrapper.find(ProposalButton).prop('disabled')).toBeUndefined();
  });

  it('Renders disabled button', () => {
    const wrapper = shallow(<ProposalSubmitButtonComponent {...disabledButtonProps} />);

    expect(wrapper.find(DisabledProposalButton)).toHaveLength(1);
    expect(wrapper.find(DisabledProposalButton).prop('disabled')).toBe(true);
  });

  it('Check a11y rules', () => {
    const wrapper = shallow(<ProposalSubmitButtonComponent />);

    expect(wrapper.find(FontAwesomeIcon).prop('aria-hidden')).toBe(true);
  });
});
