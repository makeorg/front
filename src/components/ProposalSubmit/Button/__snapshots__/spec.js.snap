// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProposalSubmitButtonComponent snapshot must match the snapshot by default 1`] = `
.c3 {
  display: none;
  visibility: hidden;
}

.c1 {
  margin-right: 0.375rem;
}

.c0 {
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  font-family: 'Roboto Condensed Bold',Arial,serif;
  font-weight: bold;
  font-size: 0.875rem;
  border: none;
  border-radius: 1.875rem;
  text-transform: uppercase;
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.5);
  padding: 0.3125rem 0.9375rem;
  color: rgb(255,255,255);
  background: rgb(118,118,118);
  background-color: rgb(118,118,118);
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.2);
  padding: 0.3125rem 0.625rem;
}

.c2 {
  margin-right: 0.375rem;
  margin: 0;
}

@media (min-width:47.94rem) {
  .c3 {
    display: block;
    visibility: visible;
  }
}

@media (min-width:47.94rem) {
  .c1 {
    margin-right: 0.625rem;
  }
}

@media (min-width:47.94rem) {
  .c0 {
    font-size: 1rem;
  }
}

@media (min-width:47.94rem) {
  .c0 {
    padding: 0.5625rem 1.25rem;
  }
}

@media (min-width:47.94rem) {
  .c2 {
    margin-right: 0.625rem;
  }
}

<button
  className="c0"
  disabled={true}
  id="proposal-submit-button"
  type="submit"
>
  <span
    className="ButtonElements__IconInButton-sc-13hw5ul-9 c1 c2"
  >
    <svg
      aria-hidden={true}
      className="svg-inline--fa fa-pencil-alt fa-w-16 "
      data-icon="pencil-alt"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </span>
  <div
    className="c3 HiddenElements__HiddenOnMobile-sc-1k80phd-1 c3"
  >
    common.propose
  </div>
</button>
`;

exports[`ProposalSubmitButtonComponent snapshot must match the snapshot if field is collapsed 1`] = `
.c3 {
  display: none;
  visibility: hidden;
}

.c1 {
  margin-right: 0.375rem;
}

.c0 {
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  font-family: 'Roboto Condensed Bold',Arial,serif;
  font-weight: bold;
  font-size: 0.875rem;
  border: none;
  border-radius: 1.875rem;
  text-transform: uppercase;
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.5);
  padding: 0.3125rem 0.9375rem;
  color: rgb(255,255,255);
  background: rgb(118,118,118);
  background-color: rgb(118,118,118);
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.2);
  padding: 0.3125rem 0.625rem;
}

.c2 {
  margin-right: 0.375rem;
  margin: 0;
}

@media (min-width:47.94rem) {
  .c3 {
    display: block;
    visibility: visible;
  }
}

@media (min-width:47.94rem) {
  .c1 {
    margin-right: 0.625rem;
  }
}

@media (min-width:47.94rem) {
  .c0 {
    font-size: 1rem;
  }
}

@media (min-width:47.94rem) {
  .c0 {
    padding: 0.5625rem 1.25rem;
  }
}

@media (min-width:47.94rem) {
  .c2 {
    margin-right: 0.625rem;
  }
}

<button
  className="c0"
  disabled={true}
  id="proposal-submit-button"
  type="submit"
>
  <span
    className="ButtonElements__IconInButton-sc-13hw5ul-9 c1 c2"
  >
    <svg
      aria-hidden={true}
      className="svg-inline--fa fa-pencil-alt fa-w-16 "
      data-icon="pencil-alt"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </span>
  <div
    className="c3 HiddenElements__HiddenOnMobile-sc-1k80phd-1 c3"
  >
    common.propose
  </div>
</button>
`;

exports[`ProposalSubmitButtonComponent snapshot must match the snapshot if field is expanded 1`] = `
.c2 {
  display: none;
  visibility: hidden;
}

.c1 {
  margin-right: 0.375rem;
}

.c0 {
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  font-family: 'Roboto Condensed Bold',Arial,serif;
  font-weight: bold;
  font-size: 0.875rem;
  border: none;
  border-radius: 1.875rem;
  text-transform: uppercase;
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.5);
  padding: 0.3125rem 0.9375rem;
  color: rgb(255,255,255);
  background: rgb(118,118,118);
  background-color: rgb(118,118,118);
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.2);
  padding: 0.3125rem 0.625rem;
}

.c3 {
  display: inline;
  visibility: visible;
}

@media (min-width:47.94rem) {
  .c2 {
    display: block;
    visibility: visible;
  }
}

@media (min-width:47.94rem) {
  .c1 {
    margin-right: 0.625rem;
  }
}

@media (min-width:47.94rem) {
  .c0 {
    font-size: 1rem;
  }
}

@media (min-width:47.94rem) {
  .c0 {
    padding: 0.5625rem 1.25rem;
  }
}

<button
  className="c0"
  disabled={true}
  id="proposal-submit-button"
  type="submit"
>
  <span
    className="c1 ButtonElements__IconInButton-sc-13hw5ul-9 c1"
  >
    <svg
      aria-hidden={true}
      className="svg-inline--fa fa-pencil-alt fa-w-16 "
      data-icon="pencil-alt"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </span>
  <span
    className="c2 c3"
  >
    common.propose
  </span>
</button>
`;

exports[`ProposalSubmitButtonComponent snapshot must match the snapshot if user can submit 1`] = `
.c3 {
  display: none;
  visibility: hidden;
}

.c1 {
  margin-right: 0.375rem;
}

.c0 {
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  font-family: 'Roboto Condensed Bold',Arial,serif;
  font-weight: bold;
  font-size: 0.875rem;
  border: none;
  border-radius: 1.875rem;
  text-transform: uppercase;
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.5);
  padding: 0.3125rem 0.9375rem;
  color: rgb(255,255,255);
  background: rgb(232,24,68);
  background-color: rgb(232,24,68);
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.5);
  padding: 0.3125rem 0.625rem;
}

.c2 {
  margin-right: 0.375rem;
  margin: 0;
}

@media (min-width:47.94rem) {
  .c3 {
    display: block;
    visibility: visible;
  }
}

@media (min-width:47.94rem) {
  .c1 {
    margin-right: 0.625rem;
  }
}

@media (min-width:47.94rem) {
  .c0 {
    font-size: 1rem;
  }
}

@media (min-width:47.94rem) {
  .c0 {
    padding: 0.5625rem 1.25rem;
  }
}

@media (min-width:47.94rem) {
  .c2 {
    margin-right: 0.625rem;
  }
}

<button
  className="c0"
  id="proposal-submit-button"
  tabIndex={0}
  type="submit"
>
  <span
    className="ButtonElements__IconInButton-sc-13hw5ul-9 c1 c2"
  >
    <svg
      aria-hidden={true}
      className="svg-inline--fa fa-pencil-alt fa-w-16 "
      data-icon="pencil-alt"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </span>
  <div
    className="c3 HiddenElements__HiddenOnMobile-sc-1k80phd-1 c3"
  >
    common.propose
  </div>
</button>
`;

exports[`ProposalSubmitButtonComponent snapshot must match the snapshot if user can't submit 1`] = `
.c3 {
  display: none;
  visibility: hidden;
}

.c1 {
  margin-right: 0.375rem;
}

.c0 {
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  font-family: 'Roboto Condensed Bold',Arial,serif;
  font-weight: bold;
  font-size: 0.875rem;
  border: none;
  border-radius: 1.875rem;
  text-transform: uppercase;
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.5);
  padding: 0.3125rem 0.9375rem;
  color: rgb(255,255,255);
  background: rgb(118,118,118);
  background-color: rgb(118,118,118);
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.2);
  padding: 0.3125rem 0.625rem;
}

.c2 {
  margin-right: 0.375rem;
  margin: 0;
}

@media (min-width:47.94rem) {
  .c3 {
    display: block;
    visibility: visible;
  }
}

@media (min-width:47.94rem) {
  .c1 {
    margin-right: 0.625rem;
  }
}

@media (min-width:47.94rem) {
  .c0 {
    font-size: 1rem;
  }
}

@media (min-width:47.94rem) {
  .c0 {
    padding: 0.5625rem 1.25rem;
  }
}

@media (min-width:47.94rem) {
  .c2 {
    margin-right: 0.625rem;
  }
}

<button
  className="c0"
  disabled={true}
  id="proposal-submit-button"
  type="submit"
>
  <span
    className="ButtonElements__IconInButton-sc-13hw5ul-9 c1 c2"
  >
    <svg
      aria-hidden={true}
      className="svg-inline--fa fa-pencil-alt fa-w-16 "
      data-icon="pencil-alt"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </span>
  <div
    className="c3 HiddenElements__HiddenOnMobile-sc-1k80phd-1 c3"
  >
    common.propose
  </div>
</button>
`;
